src/commands/branch.rs

```diff
@@ -61,7 +61,14 @@ pub struct BranchDeleteArgs {
 /// preceded by a "+". For information about branches, see
 /// https://github.com/martinvonz/jj/blob/main/docs/branches.md.
 #[derive(clap::Args, Clone, Debug)]
-pub struct BranchListArgs;
+pub struct BranchListArgs {
+    /// Show branches which local/remote targets are in the given revisions.
+    ///
+    /// If any of local/remote targets matched, the branch (including
+    /// non-matching targets) will be listed.
+    #[arg(long, short)]
+    revisions: Option<RevisionArg>,
+}
```

Comment:
(Not for this PR) I think this should also take an option that lists branches matching a glob.

Reply:
Added TODO comment about that.